pkg/config/config.go

```diff
@@ -1888,3 +1888,25 @@ func GetVectorURL(datatype DataType) (string, error) {
 	}
 	return "", nil
 }
+
+// GetTraceAgentDefaultEnv returns the default env for the trace agent
+func GetTraceAgentDefaultEnv() string {
```

Comment:
nit: now that it's in a separate function, I'd just return the value at every point that you call `defaultEnv = `. It'll make the logic a bit simpler.

e.g. line 1896 just do `return Datadog.GetString("apm_config.env")`

Reply:
We still need to log where the environment comes from in the config, I'm not sure it will make the logic simpler

Reply:
Oh you can just log before you return. That's totally fine.